tests/unit/test_log_proxy_consumer.py

```diff
@@ -274,6 +275,11 @@ def test__promtail_must_be_downloaded_in_workload_sha256_match(self):
                 ],
             )
 
+    @patch("ops.model.Container.pull")
```

Comment:
If my comment above regarding handling of relation departed event merits consideration, should there be a test case to check that the edge case mentioned there is tested for ?

Reply:
Can we avoid mocking pull here?

Reply:
We could, but since we explicitly want it to return an exception as a side effect (rather than mocking away the actual behavior of pulling a file in the harness), the alternative is just to simply use the harness `FakePebbleClient` to try to get a path which doesn't exist. I can do either, but it seems pretty innocuous here.